src/RHS.ts

```diff
@@ -35,25 +35,44 @@ export default class RHS extends Builder {
   }
 
   terminalify(parentNode: Element) {
+    // we store effects to perform later so the iteration doesn't get messed up
+    let pairs: { parent: Element; child: Text }[] = [];
     for (let i = 0; i < parentNode.childNodes.length; i++) {
       const node = parentNode.childNodes[i];
-
       if (node.nodeType === 3) {
-        const text = node.textContent!.trim();
-        const pieces = text.split(/\s/);
-
-        pieces.forEach(p => {
-          if (p.length === 0) {
-            return;
+        pairs.push({ parent: parentNode, child: node as Text });
+      } else if (
+        node.nodeType === 1 &&
+        // @ts-ignore
+        (node.tagName === 'INS' || node.tagName === 'DEL' || node.tagName === 'MARK')
```

Comment:
curious what error you're ignoring here. Is it because `node` here is typed `Node` and not `Element`? Can you use `node.nodeName` instead of `node.tagName` here?

Reply:
Yup, it's typed as ChildNode, which lacks `tagName`:

```
src/RHS.ts:46:15 - error TS2339: Property 'tagName' does not exist on type 'ChildNode'.

46         (node.tagName === 'INS' || node.tagName === 'DEL' || node.tagName === 'MARK')
```

and yup, switching to `nodeName` fixes it and works just as well.

Kind of a shame that TypeScript can't use the `nodeName === 1` to refine to `Element`.

Reply:
It _could_ if `node`'s type was a discriminated union (i.e. a closed set with `nodeType` as a discriminant property), however `Node` (and `ChildNode`) are open-ended base types and therefore could theoretically be implemented by a value that returns `1` that *isn't* an `Element`.

You *can* however use `instanceof` as a type guard: 
```ts
if (node instanceof Element && node.tagName === 'INS') {
}
```

You can also use a custom type guard:
```ts
function isElement(node: Node): node is Element {
  return node.nodeType === 1;
}
...
if (isElement(node) && node.tagName = ...
```

Its a small enough function that v8 can easily inline.

Reply:
Generally speaking I try not to add code to please the type checker when the code is already well-typed but TypeScript can't prove it; that's what `@ts-ignore` is for, right?

(It's well-typed in the sense that `nodeType` is logically intended to be a discriminant and any code which violates that has violated the type contract, even if that's not a contract TypeScript enforces.)